CIP-0XXX/README.md

```diff
@@ -0,0 +1,561 @@
+ 7    / msgReplyMessages
+ 8    / msgDone
+ 9 
+10  msgInit = [0]
+11  msgRequestMessageIds = [1, isBlocking, messageCount, messageCount]
+12  msgReplyMessageIds = [2, [ *messageIdAndSize ] ]
+13  msgRequestMessages = [3, messageIdList ]
+14  msgReplyMessages = [4, ]
+15  msgDone = [5, ]
+16
+17  isBlocking = false / true
+18  messageCount = word16
+19  messageId = bstr
+20  messageBody = bstr
+21  messageIdAndSize = [ messageId, messageSizeInBytes ]
+22  messageIdList = [ * messageId ]
+23  messageList = [ * message ]
+24  messageSizeInBytes = word32
+25  kesSignature = bstr
+26  operationalCertificate = bstr
+27  blockNumber = word32
+28  
+29  message = [
+30    messageId,
+31    messageBody,
+32    blockNumber,
+33    kesSignature,
+34    operationalCertificate
+35  ]
+30
+```
+
+### Client and server implementation
+
+This mini-protocol is designed with two goals in mind:
+* diffuse messages with high efficiency
+* protect from asymmetric resource atatckes from the message consumer against the message provider
+
+The mini-protocol is based on two pull-based operations:
+- the message consumer asks for message ids,
+- and uses these ids to request a batch of messages (which it has not received yet)
+
+The client is responsible for initiating the mini-protocol with a peer node, but the server (i.e. the other node) is the one who asks for information.
+
+The client maintains a limited size FIFO queue of outstanding messages for each of the servers it is connected to, so does the server with a mirror FIFO queue of message ids:
+- the server asks for the next message ids and acknowledges for the previous message ids received (and removed from its queue).
+- the client removes the acknowledged ids from the FIFO queue it maintains for the server.
+- the server can download the content of the messages by giving an unordered list of ids to the client.
+
+The protocol supports blocking and non-blocking requests:
+- the client must reply immediately to a non-blocking request.
+- the server must wait until the client has at least one message available.
+- if the current queue of the server is empty, it must use a blocking request and a non-blocking request otherwise.
+
+### Network load
+
+#### Mithril extra network usage
+
+> [!NOTE]
+> The computation must apply a multiplicative factor based on the number of peers connected and/or the overall redundancy of messages transmitted with the diffusion mechanism. The current value is **2**.
+
+> [!WARNING]
+> Some compression can be applied to the Mithril signatures which allows them to always be on the lower bound size, but it is not implemented yet.
+
+> [!NOTE]  
+> For a total of **3,100** Cardano SPOs on the `mainnet`, on an average **50%** of them will be eligible to send signatures (i.e. will win at least one lottery in the Mithril protocol). This means that if the full Cardano stake distribution is involved in the Mithril protocol, only **1550** signers will send signatures at each round. 
+
```

Comment:
Could: move it into the text (see other comment)

```suggestion
```

Reply:
Done